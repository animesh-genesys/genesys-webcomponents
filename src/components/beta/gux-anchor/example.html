<h1>gux-anchor-beta</h1>

<h2>Default</h2>
<gux-anchor-beta>
  <a href="#">Link</a>
</gux-anchor-beta>

<h2>Icon</h2>
<gux-anchor-beta>
  <gux-icon icon-name="message-email" decorative></gux-icon>
  <a href="#">Link Example</a>
</gux-anchor-beta>

<h2>Target</h2>
<gux-anchor-beta>
  <a href="#">Open in current window</a>
</gux-anchor-beta>

<gux-anchor-beta>
  <a href="#" target="_blank">Open in new tab</a>
</gux-anchor-beta>

<h2>External Link (Icon)</h2>
<gux-anchor-beta>
  <a href="https://spark.genesys.com/" target="_blank">External Link - Spark</a>
</gux-anchor-beta>

<h2>In Table Cell Link</h2>
<gux-table-beta>
  <table slot="data">
    <thead>
      <tr>
        <th data-column-name="first-name">First and Last Name</th>
        <th data-column-name="last-name">Other</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>John Doe</td>
        <td>
          <gux-anchor-beta table="true">
            <a href="#">John Doe Information</a>
          </gux-anchor-beta>
        </td>
      </tr>
      <tr>
        <td>Jane Doe</td>
        <td>
          <gux-anchor-beta table="true">
            <a href="#">Jane Doe Information</a>
          </gux-anchor-beta>
        </td>
      </tr>
    </tbody>
  </table>
</gux-table-beta>
