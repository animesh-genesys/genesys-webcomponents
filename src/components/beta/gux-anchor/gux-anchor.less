@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/spacing.less';

:host {
  display: flex;
  align-items: center;

  ::slotted(gux-icon) {
    padding-right: @spacing-xs;
    color: @gux-blue-60;
  }
}

:host([table='true']) {
  ::slotted(a) {
    .gux-anchor-table();
  }
}

:host {
  ::slotted(a) {
    .gux-anchor();
  }

  ::slotted(a:hover) {
    .gux-anchor-hover();
  }

  ::slotted(a:active) {
    .gux-anchor-active();
  }

  ::slotted(a:focus) {
    .gux-anchor-focus();
  }
}

gux-icon {
  padding-right: @spacing-xs;
  color: @gux-blue-60;
}
