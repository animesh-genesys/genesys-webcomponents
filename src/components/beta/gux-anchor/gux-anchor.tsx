import { Component, Element, Host, h, JSX, Prop } from '@stencil/core';
import { trackComponent } from '../../../usage-tracking';

/**
 * @slot - Anchor element
 */
@Component({
  styleUrl: 'gux-anchor.less',
  tag: 'gux-anchor-beta',
  shadow: true
})
export class GuxAnchor {
  @Element()
  private root: HTMLElement;

  /**
   * True when anchor component is used within a table cell
   */
  @Prop()
  table: boolean = false;

  componentWillLoad() {
    trackComponent(this.root);
  }

  renderExternalLinkIcon() {
    if (this.root.querySelector('a').host !== window.location.host) {
      return (
        <gux-icon icon-name="external-link" decorative></gux-icon>
      ) as JSX.Element;
    }
  }

  render(): JSX.Element {
    return (
      <Host table={this.table.toString()}>
        {this.renderExternalLinkIcon()}
        <slot />
        <slot />
      </Host>
    ) as JSX.Element;
  }
}
