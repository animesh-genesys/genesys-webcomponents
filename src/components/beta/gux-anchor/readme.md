# gux-anchor



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                            | Type      | Default |
| -------- | --------- | ------------------------------------------------------ | --------- | ------- |
| `table`  | `table`   | True when anchor component is used within a table cell | `boolean` | `false` |


## Slots

| Slot | Description    |
| ---- | -------------- |
|      | Anchor element |


## Dependencies

### Depends on

- [gux-icon](../../stable/gux-icon)

### Graph
```mermaid
graph TD;
  gux-anchor-beta --> gux-icon
  style gux-anchor-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
