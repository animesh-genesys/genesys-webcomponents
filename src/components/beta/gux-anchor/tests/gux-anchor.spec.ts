import { newSpecPage } from '@stencil/core/testing';
import { GuxAnchor } from '../gux-anchor';

const components = [GuxAnchor];
const language = 'en';

describe('gux-anchor-beta', () => {
  describe('#render', () => {
    [
      '<gux-anchor-beta><a href="#">Default Link</a></gux-anchor-beta>',
      '<gux-anchor-beta><a href="#" target="_blank">Open in new tab</a></gux-anchor-beta>',
      '<gux-anchor-beta><a href="https://spark.genesys.com/" target="_blank">External Link</a></gux-anchor-beta>',
      '<gux-anchor-beta table="true"><a href="#">Tabel Cell Link</a></gux-anchor-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
