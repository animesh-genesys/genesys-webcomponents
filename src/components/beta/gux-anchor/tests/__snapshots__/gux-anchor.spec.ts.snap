// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-anchor-beta #render should render component as expected (1) 1`] = `
<gux-anchor-beta table="false">
  <mock:shadow-root>
    <gux-icon decorative="" icon-name="external-link"></gux-icon>
    <slot></slot>
    <slot></slot>
  </mock:shadow-root>
  <a href="#">
    Default Link
  </a>
</gux-anchor-beta>
`;

exports[`gux-anchor-beta #render should render component as expected (2) 1`] = `
<gux-anchor-beta table="false">
  <mock:shadow-root>
    <gux-icon decorative="" icon-name="external-link"></gux-icon>
    <slot></slot>
    <slot></slot>
  </mock:shadow-root>
  <a href="#" target="_blank">
    Open in new tab
  </a>
</gux-anchor-beta>
`;

exports[`gux-anchor-beta #render should render component as expected (3) 1`] = `
<gux-anchor-beta table="false">
  <mock:shadow-root>
    <gux-icon decorative="" icon-name="external-link"></gux-icon>
    <slot></slot>
    <slot></slot>
  </mock:shadow-root>
  <a href="https://spark.genesys.com/" target="_blank">
    External Link
  </a>
</gux-anchor-beta>
`;

exports[`gux-anchor-beta #render should render component as expected (4) 1`] = `
<gux-anchor-beta table="true">
  <mock:shadow-root>
    <gux-icon decorative="" icon-name="external-link"></gux-icon>
    <slot></slot>
    <slot></slot>
  </mock:shadow-root>
  <a href="#">
    Tabel Cell Link
  </a>
</gux-anchor-beta>
`;
